barretenberg/cpp/src/barretenberg/stdlib/primitives/memory/ram_table.cpp

```diff
@@ -15,19 +15,20 @@
 namespace bb::stdlib {
 
 /**
- * @brief Construct a new ram table<Builder>::ram table object. It's dynamic memory!
+ * @brief Construct a new RAM table, i.e., dynamic memory with a fixed length.
  *
  * @tparam Builder
- * @param table_entries vector of field elements that will initialize the RAM table
+ * @param table_size size of the table we are constructing
  */
-template <typename Builder> ram_table<Builder>::ram_table(Builder* builder, const size_t table_size)
+template <typename Builder>
+ram_table<Builder>::ram_table(Builder* builder, const size_t table_size)
+    : length(table_size)
+    , context(builder)
 {
-    static_assert(HasPlookup<Builder>);
-    _context = builder;
-    _length = table_size;
-    _index_initialized.resize(table_size);
-    for (size_t i = 0; i < _index_initialized.size(); ++i) {
-        _index_initialized[i] = false;
+    static_assert(IsUltraOrMegaBuilder<Builder>);
```

Comment:
the natural way to do this would be a concept on the template param but honestly this can probably just be removed